---
title: "Dlab_SRA_NCBI_Submission"
author: "Matias Gomez"
date: "2025-04-15"
output: html_document
---

## Raw data in Unity cluster

First, I need to find the fastq files that correspond to each sample that made into the final version as there were at least two round of sequencing fro Dlab samples.

Now, get a list of all samples in the final raw VCF file used for SNP filtering and analyses 


Conect to cisco VPN with URI credentials anf then log in to unity cluster ans use an interactive session

```{bash}
$ ssh -l matias_gomez ssh3.hac.uri.edu
$ interactive
$ cd /data/pradalab/mgomez/Diploria/curaçao
```

see content in curaçao/snpsaurus

```{bash}
$ ls -l /data/pradalab/mgomez/Diploria/curaçao/snpsaurus/
```

```{bash}
total 8
drwxr-xr-x 2 matias_gomez pradalab 4096 Sep 18  2020 1287
drwxr-xr-x 2 matias_gomez pradalab 4096 Sep 18  2020 1292
```


So two folders 1287 and 1292 that correspond to two sequencing lanes

count files in each one

```{bash}
$ ls /data/pradalab/mgomez/Diploria/curaçao/snpsaurus/1287/*fastq.gz | wc -l
```

195
so 97 samples plus one Undetermined_S0_L005_R1_001.fastq.gz

And here files are name barcode, see for examples


```{bash}
$ ls /data/pradalab/mgomez/Diploria/curaçao/snpsaurus/1287/*fastq.gz | head
```


```{bash}
/data/pradalab/mgomez/Diploria/curaçao/snpsaurus/1287/1287_AAGAGGCA-AAGGAGTA_S107_L005_R1_001.fastq.gz
/data/pradalab/mgomez/Diploria/curaçao/snpsaurus/1287/1287_AAGAGGCA-ACTGCATA_S95_L005_R1_001.fastq.gz
/data/pradalab/mgomez/Diploria/curaçao/snpsaurus/1287/1287_AAGAGGCA-AGAGTAGA_S71_L005_R1_001.fastq.gz
/data/pradalab/mgomez/Diploria/curaçao/snpsaurus/1287/1287_AAGAGGCA-CTAAGCCT_S119_L005_R1_001.fastq.gz
/data/pradalab/mgomez/Diploria/curaçao/snpsaurus/1287/1287_AAGAGGCA-CTCTCTAT_S47_L005_R1_001.fastq.gz
/data/pradalab/mgomez/Diploria/curaçao/snpsaurus/1287/1287_AAGAGGCA-GTAAGGAG_S83_L005_R1_001.fastq.gz
/data/pradalab/mgomez/Diploria/curaçao/snpsaurus/1287/1287_AAGAGGCA-TAGATCGC_S35_L005_R1_001.fastq.gz
/data/pradalab/mgomez/Diploria/curaçao/snpsaurus/1287/1287_AAGAGGCA-TATCCTCT_S59_L005_R1_001.fastq.gz
/data/pradalab/mgomez/Diploria/curaçao/snpsaurus/1287/1287_AGGCAGAA-AAGGAGTA_S99_L005_R1_001.fastq.gz
/data/pradalab/mgomez/Diploria/curaçao/snpsaurus/1287/1287_AGGCAGAA-ACTGCATA_S87_L005_R1_001.fastq.gz
```



Okay, so far the files used as final raw VCF was diploria2.vcf

But according to file diploria2key.txt there were 4 sequencing lanes: 
run_3286 with 3 samples 
run_3285 with 24 samples; and these two runs have the code 4+colony number. 

run_1287 with 41 samples
run_1292 with 41 samples which are the same as in run run_1287 so these could all be merged. These samples follow de Dlab key instead of the 4+colony numbers used in the original and first sequenced samples in run 3285 and 3286


```{bash}
$ cat /data/pradalab/mgomez/Diploria/curaçao/diploria2archive/diploria2_key.txt
```


```{bash}
run_3286
CGAGGCTG-CTAAGCCT       439_dip
AAGAGGCA-CTAAGCCT       440_dip
CATACGTA-CTAAGCCT       462_dip
run_3285
CAGAGAGG-TAGATCGC       401_dip
GCTACGCT-TAGATCGC       410_dip
CGAGGCTG-TAGATCGC       425_dip
CAGAGAGG-CTCTCTAT       402_dip
GCTACGCT-CTCTCTAT       413_dip
CGAGGCTG-CTCTCTAT       427_dip
CAGAGAGG-TATCCTCT       403_dip
GCTACGCT-TATCCTCT       417_dip
CGAGGCTG-TATCCTCT       430_dip
CAGAGAGG-AGAGTAGA       404_dip
GCTACGCT-AGAGTAGA       420_dip
CGAGGCTG-AGAGTAGA       431_dip
CAGAGAGG-GTAAGGAG       405_dip
GCTACGCT-GTAAGGAG       421_dip
CGAGGCTG-GTAAGGAG       432_dip
CAGAGAGG-ACTGCATA       406_dip
GCTACGCT-ACTGCATA       422_dip
CGAGGCTG-ACTGCATA       435_dip
CAGAGAGG-AAGGAGTA       407_dip
GCTACGCT-AAGGAGTA       423_dip
CGAGGCTG-AAGGAGTA       437_dip
CAGAGAGG-CTAAGCCT       408_dip
GCTACGCT-CTAAGCCT       424_dip
CGAGGCTG-CTAAGCCT       438_dip
run_1287
TAAGGCGA-TAGATCGC       D_lab_01_tissue
CGTACTAG-TAGATCGC       D_lab_02_sperm
AGGCAGAA-TAGATCGC       D_lab_03_tissue
TCCTGAGC-TAGATCGC       D_lab_04_sperm
GGACTCCT-TAGATCGC       D_lab_05_sperm
TAGGCATG-TAGATCGC       D_lab_06_tissue
CTCTCTAC-TAGATCGC       D_lab_08_tissue
CAGAGAGG-TAGATCGC       D_lab_09_tissue
GCTACGCT-TAGATCGC       D_lab_10_tissue
CGAGGCTG-TAGATCGC       D_lab_11_tissue
AAGAGGCA-TAGATCGC       D_lab_12_tissue
GTAGAGGA-TAGATCGC       D_lab_13_tissue
TAAGGCGA-CTCTCTAT       D_lab_14_tissue
CGTACTAG-CTCTCTAT       D_lab_16_tissue
AGGCAGAA-CTCTCTAT       D_lab_18_tissue
TCCTGAGC-CTCTCTAT       D_lab_20_tissue
GGACTCCT-CTCTCTAT       D_lab_21_tissue
TAGGCATG-CTCTCTAT       D_lab_22_sperm
CTCTCTAC-CTCTCTAT       D_lab_23_tissue
CAGAGAGG-CTCTCTAT       D_lab_24_tissue
GCTACGCT-CTCTCTAT       D_lab_25_tissue
CGAGGCTG-CTCTCTAT       D_lab_26_tissue
AAGAGGCA-CTCTCTAT       D_lab_29_tissue
GTAGAGGA-CTCTCTAT       D_lab_30_tissue
TAAGGCGA-TATCCTCT       D_lab_32_tissue
CGTACTAG-TATCCTCT       D_lab_33_tissue
AGGCAGAA-TATCCTCT       D_lab_34_sperm
TCCTGAGC-TATCCTCT       D_lab_35_tissue
GGACTCCT-TATCCTCT       D_lab_36_tissue
TAGGCATG-TATCCTCT       D_lab_37_tissue
CTCTCTAC-TATCCTCT       D_lab_38_tissue
CAGAGAGG-TATCCTCT       D_lab_39_tissue
GCTACGCT-TATCCTCT       D_lab_A__tissue
CGAGGCTG-TATCCTCT       D_lab_C_sperm
AAGAGGCA-TATCCTCT       D_lab_E__sperm
GTAGAGGA-TATCCTCT       D_lab_K_sperm
TAAGGCGA-AGAGTAGA       D_lab_UK_1_sperm
CGTACTAG-AGAGTAGA       D_lab_UK_2_sperm
AGGCAGAA-AGAGTAGA       D_lab_UK_4_sperm
TCCTGAGC-AGAGTAGA       D_lab_UK_5_sperm
GGACTCCT-AGAGTAGA       D_lab_UK_6_sperm
run_1292
TAAGGCGA-TAGATCGC       D_lab_01_tissue
CGTACTAG-TAGATCGC       D_lab_02_sperm
AGGCAGAA-TAGATCGC       D_lab_03_tissue
TCCTGAGC-TAGATCGC       D_lab_04_sperm
GGACTCCT-TAGATCGC       D_lab_05_sperm
TAGGCATG-TAGATCGC       D_lab_06_tissue
CTCTCTAC-TAGATCGC       D_lab_08_tissue
CAGAGAGG-TAGATCGC       D_lab_09_tissue
GCTACGCT-TAGATCGC       D_lab_10_tissue
CGAGGCTG-TAGATCGC       D_lab_11_tissue
AAGAGGCA-TAGATCGC       D_lab_12_tissue
GTAGAGGA-TAGATCGC       D_lab_13_tissue
TAAGGCGA-CTCTCTAT       D_lab_14_tissue
CGTACTAG-CTCTCTAT       D_lab_16_tissue
AGGCAGAA-CTCTCTAT       D_lab_18_tissue
TCCTGAGC-CTCTCTAT       D_lab_20_tissue
GGACTCCT-CTCTCTAT       D_lab_21_tissue
TAGGCATG-CTCTCTAT       D_lab_22_sperm
CTCTCTAC-CTCTCTAT       D_lab_23_tissue
CAGAGAGG-CTCTCTAT       D_lab_24_tissue
GCTACGCT-CTCTCTAT       D_lab_25_tissue
CGAGGCTG-CTCTCTAT       D_lab_26_tissue
AAGAGGCA-CTCTCTAT       D_lab_29_tissue
GTAGAGGA-CTCTCTAT       D_lab_30_tissue
TAAGGCGA-TATCCTCT       D_lab_32_tissue
CGTACTAG-TATCCTCT       D_lab_33_tissue
AGGCAGAA-TATCCTCT       D_lab_34_sperm
TCCTGAGC-TATCCTCT       D_lab_35_tissue
GGACTCCT-TATCCTCT       D_lab_36_tissue
TAGGCATG-TATCCTCT       D_lab_37_tissue
CTCTCTAC-TATCCTCT       D_lab_38_tissue
CAGAGAGG-TATCCTCT       D_lab_39_tissue
GCTACGCT-TATCCTCT       D_lab_A__tissue
CGAGGCTG-TATCCTCT       D_lab_C_sperm
AAGAGGCA-TATCCTCT       D_lab_E__sperm
GTAGAGGA-TATCCTCT       D_lab_K_sperm
TAAGGCGA-AGAGTAGA       D_lab_UK_1_sperm
CGTACTAG-AGAGTAGA       D_lab_UK_2_sperm
AGGCAGAA-AGAGTAGA       D_lab_UK_4_sperm
TCCTGAGC-AGAGTAGA       D_lab_UK_5_sperm
GGACTCCT-AGAGTAGA       D_lab_UK_6_sperm
```


This will list all the files and directories inside run_3285.tar.gz


```{bash}
$ tar -tzvf run_3285.tar.gz
```


Let's unzip run 3285: that contains 

```{bash}
$ cd /data/pradalab/mgomez/Diploria/curaçao/raw_reads
```

```{bash}
drwxrwsr-x ej/SNP            0 2019-10-11 01:13 run_3285/
-rw-rw-r-- ej/SNP    181351474 2019-09-18 14:14 run_3285/3285_TAGGCATG-ACTGCATA_S162_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    119507367 2019-09-18 14:14 run_3285/3285_GCTACGCT-ACTGCATA_S165_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP     60444533 2019-09-18 14:14 run_3285/3285_GCTACGCT-TAGATCGC_S105_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    121199674 2019-09-18 14:14 run_3285/3285_GCTACGCT-CTAAGCCT_S189_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    146466926 2019-09-18 14:14 run_3285/3285_CGAGGCTG-AGAGTAGA_S142_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    304880251 2019-09-18 14:14 run_3285/3285_CGAGGCTG-GTAAGGAG_S154_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    282581344 2019-09-18 14:14 run_3285/3285_AAGAGGCA-TAGATCGC_S107_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    566882845 2019-09-18 14:14 run_3285/3285_GGACTCCT-AAGGAGTA_S173_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    427781606 2019-09-18 14:14 run_3285/3285_TAAGGCGA-CTAAGCCT_S181_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    518794310 2019-09-18 14:14 run_3285/3285_TAAGGCGA-ACTGCATA_S157_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    179945856 2019-09-18 14:14 run_3285/3285_GTAGAGGA-AGAGTAGA_S144_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP      3317909 2019-09-18 14:14 run_3285/3285_AGGCAGAA-CTCTCTAT_S111_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    414822558 2019-09-18 14:14 run_3285/3285_CGTACTAG-TATCCTCT_S122_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    328948239 2019-09-18 14:14 run_3285/3285_CTCTCTAC-GTAAGGAG_S151_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    512395072 2019-09-18 14:14 run_3285/3285_CGTACTAG-CTAAGCCT_S182_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    392501005 2019-09-18 14:14 run_3285/3285_TCCTGAGC-CTAAGCCT_S184_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP   2705247997 2019-09-18 14:14 run_3285/Undetermined_S0_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    328454012 2019-09-18 14:14 run_3285/3285_TAGGCATG-CTAAGCCT_S186_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    142406511 2019-09-18 14:14 run_3285/3285_CAGAGAGG-AAGGAGTA_S176_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    244456764 2019-09-18 14:14 run_3285/3285_TAGGCATG-GTAAGGAG_S150_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    433570627 2019-09-18 14:14 run_3285/3285_TAAGGCGA-GTAAGGAG_S145_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    347182518 2019-09-18 14:14 run_3285/3285_AAGAGGCA-AAGGAGTA_S179_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    215759431 2019-09-18 14:14 run_3285/3285_CAGAGAGG-TAGATCGC_S104_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    222999813 2019-09-18 14:14 run_3285/3285_GTAGAGGA-CTCTCTAT_S120_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    505655179 2019-09-18 14:14 run_3285/3285_CGTACTAG-ACTGCATA_S158_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    204292550 2019-09-18 14:14 run_3285/3285_TAGGCATG-TATCCTCT_S126_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    492979857 2019-09-18 14:14 run_3285/3285_AAGAGGCA-GTAAGGAG_S155_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    248938562 2019-09-18 14:14 run_3285/3285_GTAGAGGA-ACTGCATA_S168_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    149134172 2019-09-18 14:14 run_3285/3285_AGGCAGAA-GTAAGGAG_S147_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    483802110 2019-09-18 14:14 run_3285/3285_CTCTCTAC-AAGGAGTA_S175_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    160760005 2019-09-18 14:14 run_3285/3285_CAGAGAGG-TATCCTCT_S128_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    280301316 2019-09-18 14:14 run_3285/3285_TCCTGAGC-TAGATCGC_S100_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    217838004 2019-09-18 14:14 run_3285/3285_TAGGCATG-TAGATCGC_S102_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    284163776 2019-09-18 14:14 run_3285/3285_GGACTCCT-CTCTCTAT_S113_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    311436035 2019-09-18 14:14 run_3285/3285_AGGCAGAA-ACTGCATA_S159_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    381505475 2019-09-18 14:14 run_3285/3285_TAAGGCGA-CTCTCTAT_S109_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    373916014 2019-09-18 14:14 run_3285/3285_TCCTGAGC-TATCCTCT_S124_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    432890101 2019-09-18 14:14 run_3285/3285_GGACTCCT-ACTGCATA_S161_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    432394359 2019-09-18 14:14 run_3285/3285_AGGCAGAA-CTAAGCCT_S183_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    359037480 2019-09-18 14:14 run_3285/3285_TCCTGAGC-AGAGTAGA_S136_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    233607643 2019-09-18 14:14 run_3285/3285_GCTACGCT-GTAAGGAG_S153_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    262822830 2019-09-18 14:14 run_3285/3285_GGACTCCT-AGAGTAGA_S137_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    231563621 2019-09-18 14:14 run_3285/3285_GTAGAGGA-AAGGAGTA_S180_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    386823451 2019-09-18 14:14 run_3285/3285_AAGAGGCA-TATCCTCT_S131_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    128656339 2019-09-18 14:14 run_3285/3285_CGAGGCTG-TAGATCGC_S106_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    436424534 2019-09-18 14:14 run_3285/3285_TCCTGAGC-ACTGCATA_S160_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    444232081 2019-09-18 14:14 run_3285/3285_TAAGGCGA-TATCCTCT_S121_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    147703963 2019-09-18 14:14 run_3285/3285_CAGAGAGG-ACTGCATA_S164_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP     72856601 2019-09-18 14:14 run_3285/3285_GCTACGCT-CTCTCTAT_S117_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    462409747 2019-09-18 14:14 run_3285/3285_GGACTCCT-CTAAGCCT_S185_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP     33727511 2019-09-18 14:14 run_3285/3285_CGAGGCTG-CTCTCTAT_S118_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    198016786 2019-09-18 14:14 run_3285/3285_GTAGAGGA-TAGATCGC_S108_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    207126828 2019-09-18 14:14 run_3285/3285_TAGGCATG-AAGGAGTA_S174_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    508004134 2019-09-18 14:14 run_3285/3285_TAAGGCGA-AAGGAGTA_S169_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    231617188 2019-09-18 14:14 run_3285/3285_CTCTCTAC-TATCCTCT_S127_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    272260334 2019-09-18 14:14 run_3285/3285_TAGGCATG-AGAGTAGA_S138_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    516838001 2019-09-18 14:14 run_3285/3285_CGTACTAG-GTAAGGAG_S146_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    489438816 2019-09-18 14:14 run_3285/3285_TCCTGAGC-GTAAGGAG_S148_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP     52770392 2019-09-18 14:14 run_3285/3285_CGAGGCTG-TATCCTCT_S130_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP     77473789 2019-09-18 14:14 run_3285/3285_AGGCAGAA-TATCCTCT_S123_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    469730115 2019-09-18 14:14 run_3285/3285_TCCTGAGC-AAGGAGTA_S172_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    426727688 2019-09-18 14:14 run_3285/3285_CGTACTAG-AAGGAGTA_S170_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP     22223713 2019-09-18 14:14 run_3285/3285_CGAGGCTG-AAGGAGTA_S178_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP     57661766 2019-09-18 14:14 run_3285/3285_CAGAGAGG-CTAAGCCT_S188_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    330652242 2019-09-18 14:14 run_3285/3285_GTAGAGGA-GTAAGGAG_S156_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    414783833 2019-09-18 14:14 run_3285/3285_AAGAGGCA-ACTGCATA_S167_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    515188484 2019-09-18 14:14 run_3285/3285_TAAGGCGA-AGAGTAGA_S133_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    270341228 2019-09-18 14:14 run_3285/3285_GGACTCCT-TATCCTCT_S125_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    322012740 2019-09-18 14:14 run_3285/3285_CTCTCTAC-CTCTCTAT_S115_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP      2012299 2019-09-18 14:14 run_3285/3285_CAGAGAGG-AGAGTAGA_S140_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    217563802 2019-09-18 14:14 run_3285/3285_AAGAGGCA-CTCTCTAT_S119_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    391897811 2019-09-18 14:14 run_3285/3285_CGTACTAG-TAGATCGC_S98_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    122574283 2019-09-18 14:14 run_3285/3285_CTCTCTAC-AGAGTAGA_S139_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    298515256 2019-09-18 14:14 run_3285/3285_AGGCAGAA-AGAGTAGA_S135_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    168199135 2019-09-18 14:14 run_3285/3285_CAGAGAGG-CTCTCTAT_S116_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    273382244 2019-09-18 14:14 run_3285/3285_GCTACGCT-AAGGAGTA_S177_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    121714546 2019-09-18 14:14 run_3285/3285_CAGAGAGG-GTAAGGAG_S152_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    327346004 2019-09-18 14:14 run_3285/3285_GGACTCCT-GTAAGGAG_S149_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    341168719 2019-09-18 14:14 run_3285/3285_CGTACTAG-AGAGTAGA_S134_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    337501933 2019-09-18 14:14 run_3285/3285_GGACTCCT-TAGATCGC_S101_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    306679041 2019-09-18 14:14 run_3285/3285_AGGCAGAA-TAGATCGC_S99_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    387111916 2019-09-18 14:14 run_3285/3285_AAGAGGCA-AGAGTAGA_S143_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    401555547 2019-09-18 14:14 run_3285/3285_CGTACTAG-CTCTCTAT_S110_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP     72337073 2019-09-18 14:14 run_3285/3285_GCTACGCT-AGAGTAGA_S141_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP     69366874 2019-09-18 14:14 run_3285/3285_CGAGGCTG-ACTGCATA_S166_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    243569616 2019-09-18 14:14 run_3285/3285_CGAGGCTG-CTAAGCCT_S190_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    360635463 2019-09-18 14:14 run_3285/3285_CTCTCTAC-TAGATCGC_S103_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    124436550 2019-09-18 14:14 run_3285/3285_TAGGCATG-CTCTCTAT_S114_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    399333234 2019-09-18 14:14 run_3285/3285_AGGCAGAA-AAGGAGTA_S171_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    123468046 2019-09-18 14:14 run_3285/3285_GCTACGCT-TATCCTCT_S129_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    438666281 2019-09-18 14:14 run_3285/3285_CTCTCTAC-CTAAGCCT_S187_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    351184033 2019-09-18 14:14 run_3285/3285_AAGAGGCA-CTAAGCCT_S191_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    225298918 2019-09-18 14:14 run_3285/3285_GTAGAGGA-TATCCTCT_S132_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    553578964 2019-09-18 14:14 run_3285/3285_TAAGGCGA-TAGATCGC_S97_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    253161550 2019-09-18 14:14 run_3285/3285_CTCTCTAC-ACTGCATA_S163_L002_R1_001.fastq.gz
-rw-rw-r-- ej/SNP    172202903 2019-09-18 14:14 run_3285/3285_TCCTGAGC-CTCTCTAT_S112_L002_R1_001.fastq.gz
```


Okay so far it seems th best approach to get the list of samples in the final vcf file and then use their IDs to to find specific barcodes in every run so that i can find uique individuals:


Okay, so after checking the supplemnetary table of colonies in the paper, there 45 samples used in analysis although 12 were excluded in filtering steps and clone removal leaving only 33, but in reality 35 were used for pop gen.

the list of final 35 samples is. notice that individual 20 repeated in the SNP phylogeny

Final VCF used por pop gen was: diploria2_SNPs_2alleles_30phred_30x_50missgen_50missind_NoClones_neutral_maf05_hwe01_LD1000_35ind.recode.vcf

list of individuals:

440_dip_sorted
462_dip_sorted
401_dip_sorted
425_dip_sorted
402_dip_sorted
413_dip_sorted
427_dip_sorted
403_dip_sorted
417_dip_sorted
430_dip_sorted
420_dip_sorted
431_dip_sorted
405_dip_sorted
421_dip_sorted
432_dip_sorted
422_dip_sorted
435_dip_sorted
407_dip_sorted
423_dip_sorted
408_dip_sorted
424_dip_sorted
438_dip_sorted
D_lab_04_sperm_sorted
D_lab_09_tissue_sorted
D_lab_10_tissue_sorted
D_lab_11_tissue_sorted
D_lab_33_tissue_sorted
D_lab_34_sperm_sorted
D_lab_36_tissue_sorted
D_lab_37_tissue_sorted
D_lab_A__tissue_sorted
D_lab_C_sperm_sorted
D_lab_E__sperm_sorted
D_lab_UK_1_sperm_sorted
D_lab_UK_6_sperm_sorted



Final list of samples and barcodes and run:

Sample ID	Barcode	Run
401_dip	CAGAGAGG-TAGATCGC	run_3285
402_dip	CAGAGAGG-CTCTCTAT	run_3285
403_dip	CAGAGAGG-TATCCTCT	run_3285
405_dip	CAGAGAGG-GTAAGGAG	run_3285
407_dip	CAGAGAGG-AAGGAGTA	run_3285
408_dip	CAGAGAGG-CTAAGCCT	run_3285
413_dip	GCTACGCT-CTCTCTAT	run_3285
417_dip	GCTACGCT-TATCCTCT	run_3285
420_dip	GCTACGCT-AGAGTAGA	run_3285
421_dip	GCTACGCT-GTAAGGAG	run_3285
422_dip	GCTACGCT-ACTGCATA	run_3285
423_dip	GCTACGCT-AAGGAGTA	run_3285
424_dip	GCTACGCT-CTAAGCCT	run_3285
425_dip	CGAGGCTG-TAGATCGC	run_3285
427_dip	CGAGGCTG-CTCTCTAT	run_3285
430_dip	CGAGGCTG-TATCCTCT	run_3285
431_dip	CGAGGCTG-AGAGTAGA	run_3285
432_dip	CGAGGCTG-GTAAGGAG	run_3285
435_dip	CGAGGCTG-ACTGCATA	run_3285
438_dip	CGAGGCTG-CTAAGCCT	run_3285
440_dip	AAGAGGCA-CTAAGCCT	run_3286
462_dip	CATACGTA-CTAAGCCT	run_3286
D_lab_04_sperm	TCCTGAGC-TAGATCGC	run_1287, run_1292
D_lab_09_tissue	CAGAGAGG-TAGATCGC	run_1287, run_1292
D_lab_10_tissue	GCTACGCT-TAGATCGC	run_1287, run_1292
D_lab_11_tissue	CGAGGCTG-TAGATCGC	run_1287, run_1292
D_lab_33_tissue	CGTACTAG-TATCCTCT	run_1287, run_1292
D_lab_34_sperm	AGGCAGAA-TATCCTCT	run_1287, run_1292
D_lab_36_tissue	GGACTCCT-TATCCTCT	run_1287, run_1292
D_lab_37_tissue	TAGGCATG-TATCCTCT	run_1287, run_1292
D_lab_A__tissue	GCTACGCT-TATCCTCT	run_1287, run_1292
D_lab_C_sperm	CGAGGCTG-TATCCTCT	run_1287, run_1292
D_lab_E__sperm	AAGAGGCA-TATCCTCT	run_1287, run_1292
D_lab_UK_1_sperm	TAAGGCGA-AGAGTAGA	run_1287, run_1292
D_lab_UK_6_sperm	GGACTCCT-AGAGTAGA	run_1287, run_1292

Ok, so now make a folder to cp these files and merge the ones 

Sequence files for one submission should be in one directory. Do not create complex directory structures where you have directories within directories.
Do make a new directory for each submission. Your submission subfolder is a temporary holding area and it will be removed once the whole submission is processed.

```{bash}
$ cd /data/pradalab/mgomez/Diploria/curaçao
$ mkdir SRA_Submission_Folder_Dlab_Curacao_35_Ind
```


Now cp files here but first unzip run 3285


```{bash}
tar -xvzf raw_reads/run_3285.tar.gz
```


```{bash}
run_3285/3285_TAGGCATG-ACTGCATA_S162_L002_R1_001.fastq.gz
run_3285/3285_GCTACGCT-ACTGCATA_S165_L002_R1_001.fastq.gz
run_3285/3285_GCTACGCT-TAGATCGC_S105_L002_R1_001.fastq.gz
run_3285/3285_GCTACGCT-CTAAGCCT_S189_L002_R1_001.fastq.gz
run_3285/3285_CGAGGCTG-AGAGTAGA_S142_L002_R1_001.fastq.gz
run_3285/3285_CGAGGCTG-GTAAGGAG_S154_L002_R1_001.fastq.gz
run_3285/3285_AAGAGGCA-TAGATCGC_S107_L002_R1_001.fastq.gz
run_3285/3285_GGACTCCT-AAGGAGTA_S173_L002_R1_001.fastq.gz
run_3285/3285_TAAGGCGA-CTAAGCCT_S181_L002_R1_001.fastq.gz
run_3285/3285_TAAGGCGA-ACTGCATA_S157_L002_R1_001.fastq.gz
run_3285/3285_GTAGAGGA-AGAGTAGA_S144_L002_R1_001.fastq.gz
run_3285/3285_AGGCAGAA-CTCTCTAT_S111_L002_R1_001.fastq.gz
run_3285/3285_CGTACTAG-TATCCTCT_S122_L002_R1_001.fastq.gz
run_3285/3285_CTCTCTAC-GTAAGGAG_S151_L002_R1_001.fastq.gz
run_3285/3285_CGTACTAG-CTAAGCCT_S182_L002_R1_001.fastq.gz
run_3285/3285_TCCTGAGC-CTAAGCCT_S184_L002_R1_001.fastq.gz
run_3285/Undetermined_S0_L002_R1_001.fastq.gz
run_3285/3285_TAGGCATG-CTAAGCCT_S186_L002_R1_001.fastq.gz
run_3285/3285_CAGAGAGG-AAGGAGTA_S176_L002_R1_001.fastq.gz
run_3285/3285_TAGGCATG-GTAAGGAG_S150_L002_R1_001.fastq.gz
run_3285/3285_TAAGGCGA-GTAAGGAG_S145_L002_R1_001.fastq.gz
run_3285/3285_AAGAGGCA-AAGGAGTA_S179_L002_R1_001.fastq.gz
run_3285/3285_CAGAGAGG-TAGATCGC_S104_L002_R1_001.fastq.gz
run_3285/3285_GTAGAGGA-CTCTCTAT_S120_L002_R1_001.fastq.gz
run_3285/3285_CGTACTAG-ACTGCATA_S158_L002_R1_001.fastq.gz
run_3285/3285_TAGGCATG-TATCCTCT_S126_L002_R1_001.fastq.gz
run_3285/3285_AAGAGGCA-GTAAGGAG_S155_L002_R1_001.fastq.gz
run_3285/3285_GTAGAGGA-ACTGCATA_S168_L002_R1_001.fastq.gz
run_3285/3285_AGGCAGAA-GTAAGGAG_S147_L002_R1_001.fastq.gz
run_3285/3285_CTCTCTAC-AAGGAGTA_S175_L002_R1_001.fastq.gz
run_3285/3285_CAGAGAGG-TATCCTCT_S128_L002_R1_001.fastq.gz
run_3285/3285_TCCTGAGC-TAGATCGC_S100_L002_R1_001.fastq.gz
run_3285/3285_TAGGCATG-TAGATCGC_S102_L002_R1_001.fastq.gz
run_3285/3285_GGACTCCT-CTCTCTAT_S113_L002_R1_001.fastq.gz
run_3285/3285_AGGCAGAA-ACTGCATA_S159_L002_R1_001.fastq.gz
run_3285/3285_TAAGGCGA-CTCTCTAT_S109_L002_R1_001.fastq.gz
run_3285/3285_TCCTGAGC-TATCCTCT_S124_L002_R1_001.fastq.gz
run_3285/3285_GGACTCCT-ACTGCATA_S161_L002_R1_001.fastq.gz
run_3285/3285_AGGCAGAA-CTAAGCCT_S183_L002_R1_001.fastq.gz
run_3285/3285_TCCTGAGC-AGAGTAGA_S136_L002_R1_001.fastq.gz
run_3285/3285_GCTACGCT-GTAAGGAG_S153_L002_R1_001.fastq.gz
run_3285/3285_GGACTCCT-AGAGTAGA_S137_L002_R1_001.fastq.gz
run_3285/3285_GTAGAGGA-AAGGAGTA_S180_L002_R1_001.fastq.gz
run_3285/3285_AAGAGGCA-TATCCTCT_S131_L002_R1_001.fastq.gz
run_3285/3285_CGAGGCTG-TAGATCGC_S106_L002_R1_001.fastq.gz
run_3285/3285_TCCTGAGC-ACTGCATA_S160_L002_R1_001.fastq.gz
run_3285/3285_TAAGGCGA-TATCCTCT_S121_L002_R1_001.fastq.gz
run_3285/3285_CAGAGAGG-ACTGCATA_S164_L002_R1_001.fastq.gz
run_3285/3285_GCTACGCT-CTCTCTAT_S117_L002_R1_001.fastq.gz
run_3285/3285_GGACTCCT-CTAAGCCT_S185_L002_R1_001.fastq.gz
run_3285/3285_CGAGGCTG-CTCTCTAT_S118_L002_R1_001.fastq.gz
run_3285/3285_GTAGAGGA-TAGATCGC_S108_L002_R1_001.fastq.gz
run_3285/3285_TAGGCATG-AAGGAGTA_S174_L002_R1_001.fastq.gz
run_3285/3285_TAAGGCGA-AAGGAGTA_S169_L002_R1_001.fastq.gz
run_3285/3285_CTCTCTAC-TATCCTCT_S127_L002_R1_001.fastq.gz
run_3285/3285_TAGGCATG-AGAGTAGA_S138_L002_R1_001.fastq.gz
run_3285/3285_CGTACTAG-GTAAGGAG_S146_L002_R1_001.fastq.gz
run_3285/3285_TCCTGAGC-GTAAGGAG_S148_L002_R1_001.fastq.gz
run_3285/3285_CGAGGCTG-TATCCTCT_S130_L002_R1_001.fastq.gz
run_3285/3285_AGGCAGAA-TATCCTCT_S123_L002_R1_001.fastq.gz
run_3285/3285_TCCTGAGC-AAGGAGTA_S172_L002_R1_001.fastq.gz
run_3285/3285_CGTACTAG-AAGGAGTA_S170_L002_R1_001.fastq.gz
run_3285/3285_CGAGGCTG-AAGGAGTA_S178_L002_R1_001.fastq.gz
run_3285/3285_CAGAGAGG-CTAAGCCT_S188_L002_R1_001.fastq.gz
run_3285/3285_GTAGAGGA-GTAAGGAG_S156_L002_R1_001.fastq.gz
run_3285/3285_AAGAGGCA-ACTGCATA_S167_L002_R1_001.fastq.gz
run_3285/3285_TAAGGCGA-AGAGTAGA_S133_L002_R1_001.fastq.gz
run_3285/3285_GGACTCCT-TATCCTCT_S125_L002_R1_001.fastq.gz
run_3285/3285_CTCTCTAC-CTCTCTAT_S115_L002_R1_001.fastq.gz
run_3285/3285_CAGAGAGG-AGAGTAGA_S140_L002_R1_001.fastq.gz
run_3285/3285_AAGAGGCA-CTCTCTAT_S119_L002_R1_001.fastq.gz
run_3285/3285_CGTACTAG-TAGATCGC_S98_L002_R1_001.fastq.gz
run_3285/3285_CTCTCTAC-AGAGTAGA_S139_L002_R1_001.fastq.gz
run_3285/3285_AGGCAGAA-AGAGTAGA_S135_L002_R1_001.fastq.gz
run_3285/3285_CAGAGAGG-CTCTCTAT_S116_L002_R1_001.fastq.gz
run_3285/3285_GCTACGCT-AAGGAGTA_S177_L002_R1_001.fastq.gz
run_3285/3285_CAGAGAGG-GTAAGGAG_S152_L002_R1_001.fastq.gz
run_3285/3285_GGACTCCT-GTAAGGAG_S149_L002_R1_001.fastq.gz
run_3285/3285_CGTACTAG-AGAGTAGA_S134_L002_R1_001.fastq.gz
run_3285/3285_GGACTCCT-TAGATCGC_S101_L002_R1_001.fastq.gz
run_3285/3285_AGGCAGAA-TAGATCGC_S99_L002_R1_001.fastq.gz
run_3285/3285_AAGAGGCA-AGAGTAGA_S143_L002_R1_001.fastq.gz
run_3285/3285_CGTACTAG-CTCTCTAT_S110_L002_R1_001.fastq.gz
run_3285/3285_GCTACGCT-AGAGTAGA_S141_L002_R1_001.fastq.gz
run_3285/3285_CGAGGCTG-ACTGCATA_S166_L002_R1_001.fastq.gz
run_3285/3285_CGAGGCTG-CTAAGCCT_S190_L002_R1_001.fastq.gz
run_3285/3285_CTCTCTAC-TAGATCGC_S103_L002_R1_001.fastq.gz
run_3285/3285_TAGGCATG-CTCTCTAT_S114_L002_R1_001.fastq.gz
run_3285/3285_AGGCAGAA-AAGGAGTA_S171_L002_R1_001.fastq.gz
run_3285/3285_GCTACGCT-TATCCTCT_S129_L002_R1_001.fastq.gz
run_3285/3285_CTCTCTAC-CTAAGCCT_S187_L002_R1_001.fastq.gz
run_3285/3285_AAGAGGCA-CTAAGCCT_S191_L002_R1_001.fastq.gz
run_3285/3285_GTAGAGGA-TATCCTCT_S132_L002_R1_001.fastq.gz
run_3285/3285_TAAGGCGA-TAGATCGC_S97_L002_R1_001.fastq.gz
run_3285/3285_CTCTCTAC-ACTGCATA_S163_L002_R1_001.fastq.gz
run_3285/3285_TCCTGAGC-CTCTCTAT_S112_L002_R1_001.fastq.gz
```

move run_3285 to snpsaurus folder


```{bash}
$ mv run_3285 snpsaurus/
```

Now copy from run 3285 to SRA_Submission_Folder_Dlab_Curacao_35_Ind


```{bash}
cp snpsaurus/run_3285/
```

I’ll check the barcode pairs to ensure they correspond to the correct samples in the list you provided.

Here are the barcode pairs you listed and their corresponding run:
From the provided list:

    401_dip → CAGAGAGG-TAGATCGC (run_3285)

    402_dip → CAGAGAGG-CTCTCTAT (run_3285)

    403_dip → CAGAGAGG-TATCCTCT (run_3285)

    405_dip → CAGAGAGG-GTAAGGAG (run_3285)

    407_dip → CAGAGAGG-AAGGAGTA (run_3285)

    408_dip → CAGAGAGG-CTAAGCCT (run_3285)

    413_dip → GCTACGCT-CTCTCTAT (run_3285)

    417_dip → GCTACGCT-TATCCTCT (run_3285)

    420_dip → GCTACGCT-AGAGTAGA (run_3285)

    421_dip → GCTACGCT-GTAAGGAG (run_3285)

    422_dip → GCTACGCT-ACTGCATA (run_3285)

    423_dip → GCTACGCT-AAGGAGTA (run_3285)

    424_dip → GCTACGCT-CTAAGCCT (run_3285)

    425_dip → CGAGGCTG-TAGATCGC (run_3285)

    427_dip → CGAGGCTG-CTCTCTAT (run_3285)

    430_dip → CGAGGCTG-TATCCTCT (run_3285)

    431_dip → CGAGGCTG-AGAGTAGA (run_3285)

    432_dip → CGAGGCTG-GTAAGGAG (run_3285)

    435_dip → CGAGGCTG-ACTGCATA (run_3285)

    438_dip → CGAGGCTG-CTAAGCCT (run_3285)

    440_dip → AAGAGGCA-CTAAGCCT (run_3286)

    462_dip → CATACGTA-CTAAGCCT (run_3286)

Corresponding Fastq Files:

Here are the barcode pairs that match your files in the run_3285 directory:

    CAGAGAGG-TAGATCGC → 3285_CAGAGAGG-TAGATCGC_S104_L002_R1_001.fastq.gz

    CAGAGAGG-CTCTCTAT → 3285_CAGAGAGG-CTCTCTAT_S116_L002_R1_001.fastq.gz

    CAGAGAGG-TATCCTCT → 3285_CAGAGAGG-TATCCTCT_S128_L002_R1_001.fastq.gz

    CAGAGAGG-GTAAGGAG → 3285_CAGAGAGG-GTAAGGAG_S152_L002_R1_001.fastq.gz

    CAGAGAGG-AAGGAGTA → 3285_CAGAGAGG-AAGGAGTA_S176_L002_R1_001.fastq.gz

    CAGAGAGG-CTAAGCCT → 3285_CAGAGAGG-CTAAGCCT_S188_L002_R1_001.fastq.gz

    GCTACGCT-CTCTCTAT → 3285_GCTACGCT-CTCTCTAT_S117_L002_R1_001.fastq.gz

    GCTACGCT-TATCCTCT → 3285_GCTACGCT-TATCCTCT_S129_L002_R1_001.fastq.gz

    GCTACGCT-AGAGTAGA → 3285_GCTACGCT-AGAGTAGA_S141_L002_R1_001.fastq.gz

    GCTACGCT-GTAAGGAG → 3285_GCTACGCT-GTAAGGAG_S153_L002_R1_001.fastq.gz

    GCTACGCT-ACTGCATA → 3285_GCTACGCT-ACTGCATA_S158_L002_R1_001.fastq.gz

    GCTACGCT-AAGGAGTA → 3285_GCTACGCT-AAGGAGTA_S177_L002_R1_001.fastq.gz

    GCTACGCT-CTAAGCCT → 3285_GCTACGCT-CTAAGCCT_S185_L002_R1_001.fastq.gz

    CGAGGCTG-TAGATCGC → 3285_CGAGGCTG-TAGATCGC_S106_L002_R1_001.fastq.gz

    CGAGGCTG-CTCTCTAT → 3285_CGAGGCTG-CTCTCTAT_S118_L002_R1_001.fastq.gz

    CGAGGCTG-TATCCTCT → 3285_CGAGGCTG-TATCCTCT_S130_L002_R1_001.fastq.gz

    CGAGGCTG-AGAGTAGA → 3285_CGAGGCTG-AGAGTAGA_S136_L002_R1_001.fastq.gz

    CGAGGCTG-GTAAGGAG → 3285_CGAGGCTG-GTAAGGAG_S149_L002_R1_001.fastq.gz

    CGAGGCTG-ACTGCATA → 3285_CGAGGCTG-ACTGCATA_S160_L002_R1_001.fastq.gz

    CGAGGCTG-CTAAGCCT → 3285_CGAGGCTG-CTAAGCCT_S190_L002_R1_001.fastq.gz

The run_3286 barcodes (440_dip and 462_dip) do not have corresponding files in run_3285 since they are part of a different run.

This will correctly copy the exact files from the run_3285 folder to your destination folder /data/pradalab/mgomez/Diploria/curaçao/SRA_Submission_Folder_Dlab_Curacao_35_Ind/

```{bash}
$ cp /data/pradalab/mgomez/Diploria/curaçao/snpsaurus/run_3285/{3285_TAGGCATG-ACTGCATA_S162_L002_R1_001.fastq.gz,3285_GCTACGCT-ACTGCATA_S165_L002_R1_001.fastq.gz,3285_GCTACGCT-TAGATCGC_S105_L002_R1_001.fastq.gz,3285_GCTACGCT-CTAAGCCT_S189_L002_R1_001.fastq.gz,3285_CGAGGCTG-AGAGTAGA_S142_L002_R1_001.fastq.gz,3285_CGAGGCTG-GTAAGGAG_S154_L002_R1_001.fastq.gz,3285_AAGAGGCA-TAGATCGC_S107_L002_R1_001.fastq.gz,3285_GGACTCCT-AAGGAGTA_S173_L002_R1_001.fastq.gz,3285_TAAGGCGA-CTAAGCCT_S181_L002_R1_001.fastq.gz,3285_TAAGGCGA-ACTGCATA_S157_L002_R1_001.fastq.gz,3285_GTAGAGGA-AGAGTAGA_S144_L002_R1_001.fastq.gz,3285_AGGCAGAA-CTCTCTAT_S111_L002_R1_001.fastq.gz,3285_CGTACTAG-TATCCTCT_S122_L002_R1_001.fastq.gz,3285_CTCTCTAC-GTAAGGAG_S151_L002_R1_001.fastq.gz,3285_CGTACTAG-CTAAGCCT_S182_L002_R1_001.fastq.gz,3285_TCCTGAGC-CTAAGCCT_S184_L002_R1_001.fastq.gz,3285_TAGGCATG-CTAAGCCT_S186_L002_R1_001.fastq.gz,3285_CAGAGAGG-AAGGAGTA_S176_L002_R1_001.fastq.gz,3285_TAGGCATG-GTAAGGAG_S150_L002_R1_001.fastq.gz,3285_TAAGGCGA-GTAAGGAG_S145_L002_R1_001.fastq.gz} /data/pradalab/mgomez/Diploria/curaçao/SRA_Submission_Folder_Dlab_Curacao_35_Ind/

```


Ok, count files moved, there should be 22 samples so 44 fastq files in total but there is only 20 (appears to be single end) and because run 3286 is not present 


```{bash}
$ ls -l SRA_Submission_Folder_Dlab_Curacao_35_Ind/3285*.fastq.gz | wc -l
20
```


Checking if data is single end as no R2 fastq file present

```{bash}
$ zcat SRA_Submission_Folder_Dlab_Curacao_35_Ind/3285_AAGAGGCA-TAGATCGC_S107_L002_R1_001.fastq.gz | head -n 4

```

```{bash}
@K00337:304:HCKVCBBXY:2:1101:3072:1226 1:N:0:NAGAGGCA+NAGATCGC
CNACAAGTAAACAATACAACCAAATCGCTTTTGGGGCTTCTGTAGCGTCGAACGAACTTAACAATATTTGCGCCACATCCGCGTCTGTTTTCATGCAGAGCGTGATTGTCTTTAAATTGCCGCTATCGAGGAAACAATTCATACAAGTAGA
+
A#AFFJJJJJJJJJJJJJJJJJFFJJJJJFJJJJJJFJJJJJJJJJJJJJFJJJJJFJJJJJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJA<JJJJJJJJFJJJJFJAFA<JJFJJF<JJJJFJJJFFJJJJFFF<<
```


he header line from your gzipped FASTQ file shows that this is a paired-end read file, based on the presence of a 1 after the sequence identifier. Here's the relevant portion of the header:

@K00337:304:HCKVCBBXY:2:1101:3072:1226 1:N:0:NAGAGGCA+NAGATCGC

    The 1 indicates that this is the first read in the pair (i.e., R1).

    The + sign in the header indicates the corresponding mate is likely to be found in the next file with R2 in the filename.

Since the header includes 1, this indicates it's the first read (R1) of a paired-end sequence. Therefore, your library preparation is most likely paired-end.

Now that you've confirmed the file is paired-end for R1, you would want to look for the corresponding R2 file (which should be similar in naming, ending in R2_001.fastq.gz).

Let me know if you need further clarification or assistance with these files!


Check for unique barcode combinations


```{bash}
$ ls -l snpsaurus/run_3285/3285_CAGAGAGG-TAGATCGC*.fastq.gz
```


So for now it seems that data would have been single sequenced so I'll move forward

D_lab_04_sperm	TCCTGAGC-TAGATCGC	run_1287, run_1292
D_lab_09_tissue	CAGAGAGG-TAGATCGC	run_1287, run_1292
D_lab_10_tissue	GCTACGCT-TAGATCGC	run_1287, run_1292
D_lab_11_tissue	CGAGGCTG-TAGATCGC	run_1287, run_1292
D_lab_33_tissue	CGTACTAG-TATCCTCT	run_1287, run_1292
D_lab_34_sperm	AGGCAGAA-TATCCTCT	run_1287, run_1292
D_lab_36_tissue	GGACTCCT-TATCCTCT	run_1287, run_1292
D_lab_37_tissue	TAGGCATG-TATCCTCT	run_1287, run_1292
D_lab_A__tissue	GCTACGCT-TATCCTCT	run_1287, run_1292
D_lab_C_sperm	CGAGGCTG-TATCCTCT	run_1287, run_1292
D_lab_E__sperm	AAGAGGCA-TATCCTCT	run_1287, run_1292
D_lab_UK_1_sperm	TAAGGCGA-AGAGTAGA	run_1287, run_1292
D_lab_UK_6_sperm	GGACTCCT-AGAGTAGA	run_1287, run_1292


```{bash}
cp snpsaurus/1287/1287_TCCTGAGC-TAGATCGC*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1287/1287_CAGAGAGG-TAGATCGC*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1287/1287_GCTACGCT-TAGATCGC*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1287/1287_CGAGGCTG-TAGATCGC*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1287/1287_CGTACTAG-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1287/1287_AGGCAGAA-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1287/1287_GGACTCCT-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1287/1287_TAGGCATG-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1287/1287_GCTACGCT-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1287/1287_CGAGGCTG-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1287/1287_AAGAGGCA-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1287/1287_TAAGGCGA-AGAGTAGA*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1287/1287_GGACTCCT-AGAGTAGA*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
```

so 13 files copy from run 1287

now do the same for run 1292


```{bash}
cp snpsaurus/1292/1292_TCCTGAGC-TAGATCGC*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1292/1292_CAGAGAGG-TAGATCGC*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1292/1292_GCTACGCT-TAGATCGC*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1292/1292_CGAGGCTG-TAGATCGC*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1292/1292_CGTACTAG-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1292/1292_AGGCAGAA-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1292/1292_GGACTCCT-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1292/1292_TAGGCATG-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1292/1292_GCTACGCT-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1292/1292_CGAGGCTG-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1292/1292_AAGAGGCA-TATCCTCT*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1292/1292_TAAGGCGA-AGAGTAGA*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
cp snpsaurus/1292/1292_GGACTCCT-AGAGTAGA*fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
```


so 13 files copy from run 1292


Now, merge those fastq corresponding to these 13 samples

```{bash}
$ cd SRA_Submission_Folder_Dlab_Curacao_35_Ind/
```


```{bash}
cat 1287_AAGAGGCA-TATCCTCT_S59_L005_R1_001.fastq.gz 1292_AAGAGGCA-TATCCTCT_S253_L006_R1_001.fastq.gz > merged_AAGAGGCA-TATCCTCT_R1.fastq.gz
cat 1287_CGTACTAG-TATCCTCT_S50_L005_R1_001.fastq.gz 1292_CGTACTAG-TATCCTCT_S244_L006_R1_001.fastq.gz > merged_CGTACTAG-TATCCTCT_R1.fastq.gz
cat 1287_TAAGGCGA-AGAGTAGA_S61_L005_R1_001.fastq.gz 1292_TAAGGCGA-AGAGTAGA_S255_L006_R1_001.fastq.gz > merged_TAAGGCGA-AGAGTAGA_R1.fastq.gz
cat 1287_AGGCAGAA-TATCCTCT_S51_L005_R1_001.fastq.gz 1292_AGGCAGAA-TATCCTCT_S245_L006_R1_001.fastq.gz > merged_AGGCAGAA-TATCCTCT_R1.fastq.gz
cat 1287_GCTACGCT-TAGATCGC_S33_L005_R1_001.fastq.gz 1292_GCTACGCT-TAGATCGC_S227_L006_R1_001.fastq.gz > merged_GCTACGCT-TAGATCGC_R1.fastq.gz
cat 1287_TAGGCATG-TATCCTCT_S54_L005_R1_001.fastq.gz 1292_TAGGCATG-TATCCTCT_S248_L006_R1_001.fastq.gz > merged_TAGGCATG-TATCCTCT_R1.fastq.gz
cat 1287_CAGAGAGG-TAGATCGC_S32_L005_R1_001.fastq.gz 1292_CAGAGAGG-TAGATCGC_S226_L006_R1_001.fastq.gz > merged_CAGAGAGG-TAGATCGC_R1.fastq.gz
cat 1287_GCTACGCT-TATCCTCT_S57_L005_R1_001.fastq.gz 1292_GCTACGCT-TATCCTCT_S251_L006_R1_001.fastq.gz > merged_GCTACGCT-TATCCTCT_R1.fastq.gz
cat 1287_TCCTGAGC-TAGATCGC_S28_L005_R1_001.fastq.gz 1292_TCCTGAGC-TAGATCGC_S222_L006_R1_001.fastq.gz > merged_TCCTGAGC-TAGATCGC_R1.fastq.gz
cat 1287_CGAGGCTG-TAGATCGC_S34_L005_R1_001.fastq.gz 1292_CGAGGCTG-TAGATCGC_S228_L006_R1_001.fastq.gz > merged_CGAGGCTG-TAGATCGC_R1.fastq.gz
cat 1287_GGACTCCT-AGAGTAGA_S65_L005_R1_001.fastq.gz 1292_GGACTCCT-AGAGTAGA_S259_L006_R1_001.fastq.gz > merged_GGACTCCT-AGAGTAGA_R1.fastq.gz
cat 1287_CGAGGCTG-TATCCTCT_S58_L005_R1_001.fastq.gz 1292_CGAGGCTG-TATCCTCT_S252_L006_R1_001.fastq.gz > merged_CGAGGCTG-TATCCTCT_R1.fastq.gz
cat 1287_GGACTCCT-TATCCTCT_S53_L005_R1_001.fastq.gz 1292_GGACTCCT-TATCCTCT_S247_L006_R1_001.fastq.gz > merged_GGACTCCT-TATCCTCT_R1.fastq.gz

```
    

count merged files

```{bash}
$ ls merged*fastq.gz | wc -l
13
```

Alos, since i dont have access to run 3286 where samples 440_dip	AAGAGGCA-CTAAGCCT	 462_dip	CATACGTA-CTAAGCCT	came from, i'm going to add these samples from

run 1287 and 1292

sample 440 is Dlab colony 40 and 462 is Dlab 12 but only D_lab_12_tissue	AAGAGGCA-TAGATCGC	run_1287, run_1292 was sequenced in 1287 and 1292

```{bash}
$ cp snpsaurus/1287/1287_AAGAGGCA-TAGATCGC_S35_L005_R1_001.fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
$ cp snpsaurus/1292/1292_AAGAGGCA-TAGATCGC_S229_L006_R1_001.fastq.gz SRA_Submission_Folder_Dlab_Curacao_35_Ind
```
 and merge them
 
```{bash}
$ cd SRA_Submission_Folder_Dlab_Curacao_35_Ind
cat 1287_AAGAGGCA-TAGATCGC_S35_L005_R1_001.fastq.gz 1292_AAGAGGCA-TAGATCGC_S229_L006_R1_001.fastq.gz  > merged_AAGAGGCA-TAGATCGC_R1.fastq.gz
```
 




